File: src/main/java/com/wesker926/leetcode/algorithms/p0458/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/25
// @description: 维度（进制）计算
package s1

import "math"

func poorPigs(buckets int, minutesToDie int, minutesToTest int) int {
	n := (minutesToTest / minutesToDie) + 1
	y := math.Log2(float64(buckets)) / math.Log2(float64(n))
	if y == math.Trunc(y) {
		return int(y)
	}
	return int(y) + 1
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0458/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0458.s1;

/**
 * @author wesker.gh
 * @date 2021/11/25
 * @description 维度（进制）计算
 */
public class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        int n = minutesToTest / minutesToDie + 1;
        double y = Math.log(buckets) / Math.log(n);
        int intY = (int) y;
        return y == intY ? intY : intY + 1;
    }
}
